controllers/horizon_controller.go

```diff
@@ -340,6 +345,29 @@ func (r *HorizonReconciler) reconcileNormal(ctx context.Context, instance *horiz
 		return ctrl.Result{}, err
 	}
 
+	// Create Memcached instance if no shared instance exists.
+	if instance.Spec.SharedMemcached == "" {
+		memcached := r.renderMemcached(instance)
+		op, err := controllerutil.CreateOrPatch(ctx, helper.GetClient(), memcached, func() error {
+			memcached.Labels = map[string]string{"service": instance.Name}
```

Comment:
// CreateOrPatch creates or patches the given object in the Kubernetes
// cluster. The object's desired state must be reconciled with the before
// state inside the passed in callback MutateFn.

Reply:
I guess I should be doing `memcached := memcachedv1.Memcached{}`. Rather than using that `renderMemcached` function in this scenario. Then set each of the fields within the `MutateFn()`.

I'll change it tomorrow and update accordingly.